### DISCUSSION TITLE
Configurable dashboard with multiple instances of a single palette item?

### DISCUSSION THREAD

**Discussion by RyanFrench** [cloudscape-design/components#2104] [Q&A] [answered]
Hi,

I'm currently working on building a drag and drop dashboard that allows users to add widgets and customise them. The number of widgets is limitless and the content of each widget is dynamic based on configuration the user enters. So far I've created a single Palette item and widget which can be dragged and dropped on to the dashboard, but when I try and add a second widget from the palette item I get an error 

![Screenshot 2024-03-28 at 17 04 20](https://github.com/cloudscape-design/components/assets/979780/3d0e208c-7425-4edf-91e9-3fa203b63a75)

If I create a second palette item with exactly the same details I can add both to the board, I just can't add multiple from the same palette item. Is this expected? Is there no way to have a dynamic number of widgets based on the same palette item?

**Comment by connorlanigan** [comment, marked as answer]
This issue very likely happens because the duplicate widgets on your board share the same ID. This is not supported:
> `id (string)` - the unique item identifier. The IDs of any two items in a page must be different.
>
> \- https://cloudscape.design/components/board/?tabId=api

If you want to add another widget to the board, it needs to have a different ID. One possible option is to generate a new ID for each `addedItem` in the Board component's `onItemsChange` handler.

**Reply by RyanFrench** [reply]
Thank you, you are correct. I managed to get around this by generating a new set of IDs and deep cloning the widgets each time I retrieve the dashboard widgets to add to the board. This allows me to generate more than 1 of each type of widget.
